### PR TITLE
API beatmapset lookup endpoint

### LINKED ISSUE #2625: Retrieving beatmapset with just a beatmap's ID
The response from the API's `/beatmaps/:beatmap_id` endpoint currently returns a beatmap's set incompletely.

image

It's missing the list of the beatmaps in the set for starters. This is currently a roadblock in looking up beatmaps by ID in the client (from chat links etc.), since beatmaps need to be displayed as part of a set. Should sets be looked up by a second request using the `beatmapset_id` of the response or is the entire set supposed to be included in this response?

### PR DESCRIPTION
**Pull request by nekodex**
Similar to beatmap lookup, currently only supports looking up by `beatmap_id`.

e.g.: `/api/v2/beatmapsets/lookup?beatmap_id=660345`

Ideally this functionality will eventually be merged into search, along with the ability to search other beatmap/set metadata, etc.

fixes #2625

### CODE REVIEW

File: routes/web.php [hunk 1]
```diff
@@ -255,34 +255,36 @@
...
             // POST /api/v2/chat/messages/new
-            Route::post('messages', ['uses' => 'ChatController@postMessage']);
+            Route::post('messages', 'ChatController@postMessage');
         });
 
         Route::resource('rooms', 'RoomsController', ['only' => ['show']]);
 
         Route::group(['prefix' => 'beatmapsets'], function () {
-            Route::get('favourites', ['uses' => 'BeatmapsetsController@favourites']);     //  GET /api/v2/beatmapsets/favourites
+            Route::get('favourites', 'BeatmapsetsController@favourites');     //  GET /api/v2/beatmapsets/favourites
```
Comment (nanaya) [inline routes/web.php]: might as well merge with the `// Beatmapsets` group below while at it?

Comment (nekodex) [reply routes/web.php]: hrm I wonder if that endpoint should be nested under `users` or `me` instead?

Comment (nanaya) [reply routes/web.php]: hrm right :confused:

Comment (nekodex) [reply routes/web.php]: Actually that endpoint probably isn't even needed anymore, I'll make a new PR removing that stuff.

**Pull request merged**